### DISCUSSION TITLE
Help with getting user ID from emotes and own bot

### DISCUSSION THREAD

**Discussion by CyberDruga** [disgoorg/disgo#325] [Help] [answered]
Hello,

I'm creating a bot that has to manipulate emotes in a discord server, and I want it to be able to delete every emote the bot has created and nothing else. I thought of getting the userId from the bot and pair it with the userId of the emote creator to determine if I should delete it or not, but I'm having problems getting both of those.

For the bot's userId I have no idea how to get it, and for the emotes, the information about the creator comes as nil. I'd like to know how can I get both ids. 

For the bot configuration I'm doing just this: 
```golang
	botClient, err = disgo.New(config["token"])

	botClient.OpenGateway(context.TODO())

	defer botClient.Close(context.TODO())

	session = botClient.Rest()
```

and I'm getting the emojis like this: 
```golang
	emojis, err := session.GetEmojis(snowflake.MustParse(config["server_id"]))
	if err != nil {
		return
	}

```

**Comment by topi314** [comment, marked as answer]
You can get the bot id by calling `botClient.ApplicationID()`.

As for the empty creator there seems to be a miss named json tag.
I just pushed a new [commit](https://github.com/disgoorg/disgo/commit/0162c707df18af369a795b85ae7b95530b605475) to master with a fix for this
you can run `go get github.com/disgoorg/disgo@0162c70` to get that
we will also include it in the next release

**Reply by CyberDruga** [reply]
That worked wonderfully. Thank you for the fast response, and thank you for the project. It's helping me a lot.

**Reply by topi314** [reply]
released in https://github.com/disgoorg/disgo/releases/tag/v0.16.12
